### ISSUE TITLE
Command-line arguments to Windows exe - double quotes not passed as expected

### ISSUE THREAD

**Issue by lukebakken** [Microsoft/BashOnWindows#2411]
* Your Windows build number:

`Microsoft Windows [Version 10.0.15063]`

* What you're doing and what's happening:

I first discovered this while trying out the RabbitMQ build system using WSL and the Win32 build of Erlang 20, but simplified it to this case that uses `EchoArgs.exe` (installed via `choco install echoargs`):

```
lbakken@DSCH-WIN:~$ /mnt/c/ProgramData/chocolatey/lib/echoargs/tools/EchoArgs.exe '"foobar"'
Unable to translate current working directory. Using C:\Users\lbakken
Arg 0 is <foobar>

Command line:
C:\ProgramData\chocolatey\lib\echoargs\tools\EchoArgs.exe "foobar"

lbakken@DSCH-WIN:~$ /mnt/c/ProgramData/chocolatey/lib/echoargs/tools/EchoArgs.exe '\"foobar\"'
Unable to translate current working directory. Using C:\Users\lbakken
Arg 0 is <"foobar">

Command line:
C:\ProgramData\chocolatey\lib\echoargs\tools\EchoArgs.exe \"foobar\"
```

* What's wrong / what should be happening instead:

I would expect that the argument `'"foobar"'` be passed to the Windows exe and interpreted by whatever functions parse `argv` as `"foobar"`. Instead, the Windows exe "sees" `foobar` without the double quotes. Cygwin and Msys have both solved this issue I'm guessing by adding `\` to embedded double quotes before starting the Windows exe. As you can see above, I can work around this issue for now by using `\"` instead of `"` in my command arguments.

Let me know if you'd like `strace` output or not. Thanks!

**Comment by lukebakken** [comment]
This is probably a duplicate of or related to #2164 and #1625

**Comment by benhillis** [comment]
@lukebakken - Indeed, this is a duplicate of #1625, there is a fix inbound.
